fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -872,6 +888,15 @@ public Boolean eval(@Nonnull FDBRecordStoreBase<?> store, @Nonnull EvaluationCon
             } else if (comparand == COMPARISON_SKIPPED_BINDING) {
                 return Boolean.TRUE;
             } else {
+                if (type == Type.IN) {
+                    if (listComparison == null) {
```

Comment:
I believe this will do the wrong thing if one evaluates the same comparison with different `EvaluationContext`s. In particular, there's no guarantee that the list comparand hasn't changed.

Perhaps what we really want is a `SetComparison` that can perform `IN` comparisons, and then `IN` would use that? Then I don't think you need to cache the `ListComparison` here (which I assume is being done to avoid recreating the set), and I think the logic in `ListComparison` can remove some `if` blocks.